metadata_schema.py

```diff
@@ -0,0 +1,84 @@
+import traceback
+
+from dict_util import ImmutableDict, find_values_with_key
+
+
+class MetadataSchema:
+    def __init__(self, raw_json: dict, schema_path: str):
+        self.raw_json = raw_json
+        self.schema_path = schema_path
+        self.json = raw_json
+
+    def get_json_schema(self, version_map: dict, schema_base_url: str):
+        self.json = self.insert_id_url(version_map, schema_base_url)
+        self.json = self.update_refs_urls(version_map, schema_base_url)
+        return self.json
+
+    def insert_id_url(self, version_map, schema_base_url):
+        id = generate_id(self.schema_path, schema_base_url, version_map)
+        id_key = self.get_id_key()
+        id_obj = {id_key: id}
+        json_schema = ImmutableDict(self.raw_json).insert_obj_at(id_obj, 1)
+        return json_schema
+
+    def update_refs_urls(self, version_map, schema_base_url) -> dict:
```

Comment:
Yeah, it’s a function from the original code (deleted file in PR because I renamed it). I haven't touched it yet, it's very hard to understand. We could refactor it later.